### PR TITLE
Add timers

### PR DESCRIPTION
**Pull request by harshach**
### Describe your changes :

I worked on the ..... because ...

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

**Review by ayush-shah — APPROVED** [review]

File: ingestion/src/metadata/ingestion/ometa/utils.py [hunk 1]
```diff
@@ -71,3 +74,18 @@ def _(arg) -> str:
...
     """
     return str(arg.__root__)
+
+
+def timer(method):
+    def timed(*args, **kw):
+        ts = time.time()
+        result = method(*args, **kw)
+        te = time.time()
+        if "log_time" in kw:
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/ometa/utils.py:84]: What is the usage of this? Are we checking the kwargs of the actual function? 🤔

Comment (pmbrull) [reply ingestion/src/metadata/ingestion/ometa/utils.py:84]: Also @harshach , for performance checks usually time.perf_counter() is preferred https://stackoverflow.com/questions/66036844/time-time-or-time-perf-counter-which-is-faster

File: ingestion/src/metadata/ingestion/ometa/utils.py [hunk 2]
```diff
@@ -23,6 +24,8 @@
 
 T = TypeVar("T", bound=BaseModel)  # pylint: disable=invalid-name
 
+logger: logging.Logger = logging.getLogger(__name__)
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/ometa/utils.py:27]: We could also move the timer decorator to is own module and init there a logger with
`logger: logging.Logger = logging.getLogger("Timer")`, which might be helpful to locate the logs

File: ingestion/src/metadata/ingestion/source/sql_source.py [hunk 3]
```diff
@@ -324,13 +339,15 @@ def fetch_tables(
...
 
+    @timer
     def fetch_views(
         self, inspector: Inspector, schema: str
     ) -> Iterable[OMetaDatabaseAndTable]:
         """
         Get all views in the SQL schema and prepare
         Database & Table OpenMetadata Entities
         """
+        start = time.perf_counter()
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/sql_source.py:350]: this line can be removed

**Pull request closed without merging**